File: app/src/main/java/com/github/sdpcoachme/GreetingActivity.java
package com.github.sdpcoachme;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import com.github.sdpcoachme.firebase.auth.FirebaseUIActivity;

public class GreetingActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_greeting);

        TextView receiverMessage = findViewById(R.id.message);

        Intent intent = getIntent();
        String str = intent.getStringExtra("name");

        receiverMessage.setText(str);
    }
}

File: app/src/main/java/com/github/sdpcoachme/GreetingActivity.kt
package com.github.sdpcoachme

import android.content.Intent
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material.Button
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.platform.testTag
import com.github.sdpcoachme.ui.theme.CoachMeTheme

class GreetingActivity : ComponentActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            CoachMeTheme {
                intent.getStringExtra("name")?.let { Greeting(it) }
            }
        }
    }
}

@Composable
fun Greeting(name: String) {
    val context = LocalContext.current
    Column(
        modifier = Modifier.fillMaxSize(),
        verticalArrangement = Arrangement.Center,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Text(modifier = Modifier.testTag("text"), text = "Hello $name!")
        Button(onClick = {
            val intent = Intent(context, EditProfileActivity::class.java)
            intent.putExtra("email", "fake@email.com")
            context.startActivity(intent)
        }) {
            Text(text = "Edit profile")
        }
    }
}